value/reflectcache.go

```diff
@@ -363,3 +363,101 @@ func (c *typeReflectCache) update(updates reflectCacheMap) {
 	}
 	c.value.Store(newCacheMap)
 }
+
+// Below json Unmarshal is fromk8s.io/apimachinery/pkg/util/json
+// to handle number conversions as expected by Kubernetes
+
+// limit recursive depth to prevent stack overflow errors
+const maxDepth = 10000
+
+// Unmarshal unmarshals the given data
+// If v is a *map[string]interface{}, numbers are converted to int64 or float64
+func Unmarshal(data []byte, v interface{}) error {
```

Comment:
keep private for now? I'd prefer not to pick up new callers here if we don't need to

Reply:
can you also bring over the unit tests around this?

Reply:
Done and done!